### PR TITLE
[MWRAPPER-129][MWRAPPER-128] Drop legacy stuff, require Maven 3.6.3+

### PR DESCRIPTION
**Pull request by cstamas**
Also refresh and reshuffle to not duplicate dependencies and versions. Tend to use properties for versions where there is 2+ of them, otherwise is just fluff.

---

https://issues.apache.org/jira/browse/MWRAPPER-128
https://issues.apache.org/jira/browse/MWRAPPER-129

### CODE REVIEW

**Review by jorsol — COMMENTED** [review]
The org.codehaus.plexus:plexus-io dependency seems that is not used.

File: maven-wrapper-plugin/pom.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@ under the License.
   <description>The Maven Wrapper Plugin is a plugin that provides support for the Maven Wrapper by unpacking Maven Wrapper Distribution to the current project.</description>
 
   <prerequisites>
-    <maven>${mavenVersion}</maven>
+    <maven>3.2.5</maven>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply maven-wrapper-plugin/pom.xml:37]: Yeah, I did not want to change prerequisite, it was more to kill this antipattern to "compile against eons old" that prevents noticing what is being deprecated lately in APIs and Core... (and replaced eon old aether w/ maven-resolver)

[the remark this answers is not in the document]
Comment (slawekjaranowski) [reply maven-wrapper-plugin/pom.xml:37]: I'm still not sure about it ... when we depend on newer version we can have some issues in runtime which is not cover by ITs. 
But from other side we have more information about depractions ...

The perfect way will be for me depends on the same version as in in prerequisites ... but we block using plugin for older Maven ....

So we need a compromise on it 😄

[the remark this answers is not in the document]
Comment (cstamas) [reply maven-wrapper-plugin/pom.xml:37]: Yeah, but _all plugin builds_ use at least two Maven version: 3.9.6 and ... usually 3.6.3. So what "issue in runtime" you expect? Okay, if test coverage is poor... but then we fix and improve coverage, and life goes on...

**Review by slawekjaranowski — APPROVED** [review]

File: maven-wrapper-plugin/pom.xml (hunk 1 above)
Comment (cstamas) [inline maven-wrapper-plugin/pom.xml:37]:
```suggestion
    <maven>3.6.3</maven>
```

**Review by slawekjaranowski — APPROVED** [review]
Please also add to commit message and issue title, that we also require 3.6.3 to be clear in release notes

**Comment by cstamas** [comment]
Tomorrow will check is plexus-io really used or not....

**Comment by jorsol** [comment]
> Tomorrow will check is plexus-io really used or not....

Maybe is used transitively, so never mind.

**Comment by cstamas** [comment]
Hm, removed it and it builds...

**Comment by cstamas** [comment]
Yes, is used:
```
[INFO] --- toolbox:0.1.3:tree (default-cli) @ maven-wrapper-plugin ---
[INFO] org.apache.maven.plugins:maven-wrapper-plugin:jar:3.2.1-SNAPSHOT
[INFO] +- javax.inject:javax.inject:jar:1 [compile]
[INFO] +- org.codehaus.plexus:plexus-archiver:jar:4.9.2 [compile]
[INFO] |  +- org.codehaus.plexus:plexus-io:jar:3.4.2 [compile]
[INFO] |  +- org.apache.commons:commons-compress:jar:1.26.1 [compile]
[INFO] |  +- commons-codec:commons-codec:jar:1.16.1 [compile]
[INFO] |  +- org.iq80.snappy:snappy:jar:0.4 [compile]
[INFO] |  +- org.tukaani:xz:jar:1.9 [runtime]
[INFO] |  \- com.github.luben:zstd-jni:jar:1.5.5-11 [runtime]
[INFO] +- commons-io:commons-io:jar:2.16.1 [compile]
[INFO] \- org.apache.maven.shared:maven-shared-utils:jar:3.4.2 [compile]
```

**Comment by jira-importer** [comment]
Resolve #289

**Comment by jira-importer** [comment]
Resolve #289

**Pull request merged**